File: KargoPaylasimJava/app/src/main/java/com/example/kargopaylasimjava/service/CargoApi.java
package com.example.kargopaylasimjava.service;


import com.example.kargopaylasimjava.dto.ApiResp;
import com.example.kargopaylasimjava.dto.AddressDtos;
import com.example.kargopaylasimjava.dto.AuthDtos;
import com.example.kargopaylasimjava.dto.ReceiverDtos;
import com.example.kargopaylasimjava.dto.ShipmentDtos;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface CargoApi {

    // Auth
    @POST("user_register.php")
    Call<ApiResp<AuthDtos.RegisterResp>> register(@Body AuthDtos.RegisterReq req);

    @POST("user_login.php")
    Call<ApiResp<AuthDtos.LoginResp>> login(@Body AuthDtos.LoginReq req);

    @GET("user_me.php")
    Call<ApiResp<AuthDtos.UserMeResp>> me();

    // Address
    @POST("address_create.php")
    Call<ApiResp<AddressDtos.AddressIdResp>> addressCreate(@Body AddressDtos.AddressCreateReq req);

    @POST("address_update.php")
    Call<ApiResp<Object>> addressUpdate(@Body AddressDtos.AddressUpdateReq req);

    @POST("address_detail_post.php")
    Call<ApiResp<AddressDtos.AddressDto>> addressDetailPost(@Body AddressDtos.AddressDetailReq req);

    @GET("address_list.php")
    Call<ApiResp<AddressDtos.AddressListResp>> addressList();

    @POST("address_delete.php")
    Call<ApiResp<Object>> addressDelete(@Body AddressDtos.AddressDeleteReq req);

    @GET("address_detail.php")
    Call<ApiResp<AddressDtos.AddressDto>> addressDetail(@Query("id") int id);

    @POST("address_set_default.php")
    Call<ApiResp<Object>> addressSetDefault(@Body AddressDtos.AddressSetDefaultReq req);

    // Receiver
    @POST("receiver_lookup.php")
    Call<ApiResp<ReceiverDtos.ReceiverLookupResp>> receiverLookup(@Body ReceiverDtos.ReceiverLookupReq req);

    // Shipment
    @POST("shipment_delete.php")
    Call<ApiResp<ShipmentDtos.ShipmentDeleteResp>> shipmentDelete(@Body ShipmentDtos.ShipmentDeleteReq req);

    @POST("shipment_create.php")
    Call<ApiResp<ShipmentDtos.ShipmentCreateResp>> shipmentCreate(@Body ShipmentDtos.ShipmentCreateReq req);

    @GET("shipment_list.php")
    Call<ApiResp<ShipmentDtos.ShipmentListResp>> shipmentList(@Query("type") String type);

    @GET("shipment_detail.php")
    Call<ApiResp<ShipmentDtos.ShipmentDetailResp>> shipmentDetail(@Query("id") int id);

    @POST("shipment_cancel.php")
    Call<ApiResp<Object>> shipmentCancel(@Body ShipmentDtos.ShipmentCancelReq req);

    @POST("shipment_regenerate_code.php")
    Call<ApiResp<ShipmentDtos.ShipmentRegenerateResp>> shipmentRegenerate(@Body ShipmentDtos.ShipmentRegenerateReq req);
}

File: KargoPaylasimKotlin/app/src/main/java/com/example/kargopaylasimkotlin/service/CargoApi.kt
package com.example.kargopaylasimkotlin.service

import com.example.kargopaylasimkotlin.dto.AddressCreateReq
import com.example.kargopaylasimkotlin.dto.AddressDeleteReq
import com.example.kargopaylasimkotlin.dto.AddressDetailReq
import com.example.kargopaylasimkotlin.dto.AddressDto
import com.example.kargopaylasimkotlin.dto.AddressIdResp
import com.example.kargopaylasimkotlin.dto.AddressListResp
import com.example.kargopaylasimkotlin.dto.AddressSetDefaultReq
import com.example.kargopaylasimkotlin.dto.AddressUpdateReq
import com.example.kargopaylasimkotlin.dto.ApiResp
import com.example.kargopaylasimkotlin.dto.LoginReq
import com.example.kargopaylasimkotlin.dto.LoginResp
import com.example.kargopaylasimkotlin.dto.ReceiverLookupReq
import com.example.kargopaylasimkotlin.dto.ReceiverLookupResp
import com.example.kargopaylasimkotlin.dto.RegisterReq
import com.example.kargopaylasimkotlin.dto.RegisterResp
import com.example.kargopaylasimkotlin.dto.ShipmentCancelReq
import com.example.kargopaylasimkotlin.dto.ShipmentCreateReq
import com.example.kargopaylasimkotlin.dto.ShipmentCreateResp
import com.example.kargopaylasimkotlin.dto.ShipmentDeleteReq
import com.example.kargopaylasimkotlin.dto.ShipmentDeleteResp
import com.example.kargopaylasimkotlin.dto.ShipmentDetailResp
import com.example.kargopaylasimkotlin.dto.ShipmentListResp
import com.example.kargopaylasimkotlin.dto.ShipmentRegenerateReq
import com.example.kargopaylasimkotlin.dto.ShipmentRegenerateResp
import com.example.kargopaylasimkotlin.dto.UserMeResp
import okhttp3.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.POST
import retrofit2.http.Query

interface CargoApi {

    // Auth
    @POST("user_register.php")
    suspend fun register(@Body req: RegisterReq): ApiResp<RegisterResp>

    @POST("user_login.php")
    suspend fun login(@Body req: LoginReq): ApiResp<LoginResp>

    @GET("user_me.php")
    suspend fun me(): ApiResp<UserMeResp>

    @POST("address_create.php")
    suspend fun addressCreate(@Body req: AddressCreateReq): ApiResp<AddressIdResp>

    @POST("address_update.php")
    suspend fun addressUpdate(@Body req: AddressUpdateReq): ApiResp<Any>

    @POST("address_detail_post.php")
    suspend fun addressDetailPost(@Body req: AddressDetailReq): ApiResp<AddressDto>

    @GET("address_list.php")
    suspend fun addressList(): ApiResp<AddressListResp>


    @POST("address_delete.php")
    suspend fun addressDelete(@Body req: AddressDeleteReq): ApiResp<Any>

    @GET("address_detail.php")
    suspend fun addressDetail(@Query("id") id: Int): ApiResp<AddressDto>

    @POST("address_set_default.php")
    suspend fun addressSetDefault(@Body req: AddressSetDefaultReq): ApiResp<Any>

    @POST("receiver_lookup.php")
    suspend fun receiverLookup(@Body req: ReceiverLookupReq): ApiResp<ReceiverLookupResp>

    @POST("shipment_delete.php")
    suspend fun shipmentDelete(
        @Body req: ShipmentDeleteReq
    ): ApiResp<ShipmentDeleteResp>

    @POST("shipment_create.php")
    suspend fun shipmentCreate(@Body req: ShipmentCreateReq): ApiResp<ShipmentCreateResp>

    @GET("shipment_list.php")
    suspend fun shipmentList(@Query("type") type: String = "all"): ApiResp<ShipmentListResp>

    @GET("shipment_detail.php")
    suspend fun shipmentDetail(@Query("id") id: Int): ApiResp<ShipmentDetailResp>

    @POST("shipment_cancel.php")
    suspend fun shipmentCancel(@Body req: ShipmentCancelReq): ApiResp<Any>

    @POST("shipment_regenerate_code.php")
    suspend fun shipmentRegenerate(@Body req: ShipmentRegenerateReq): ApiResp<ShipmentRegenerateResp>

}

